erts/emulator/beam/erl_proc_sig_queue.h

```diff
@@ -0,0 +1,661 @@
+/*
+ * %CopyrightBegin%
+ * 
+ * Copyright Ericsson AB 2018. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+ * %CopyrightEnd%
+ */
+
+/*
+ * Description:	Process signal queue implementation.
+ *
+ *              Currently the following signals are handled:
+ *              - Messages
+ *              - Exit
+ *              - Monitor
+ *              - Demonitor
+ *              - Monitor down
+ *              - Persistent monitor message
+ *              - Link
+ *              - Unlink
+ *              - Group leader
+ *              - Trace change
+ *
+ *              The signal queue consists of three parts:
+ *              - Outer queue (sig_inq field in process struct)
+ *              - Middle queue (sig_qs field in process struct)
+ *              - Inner queue (sig_qs field in process struct)
```

Comment:
You named the same field twice here. Shouldn't they be different fields?

Reply:
No this is correct. Both the middle and inner queues are private to the process and also share some parts which is the reason why they are in the same field. See the ErtsSignalPrivQueues type in erl_message.h for details.